test/hack/resource/parse-redirects/main.go

```diff
@@ -0,0 +1,127 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type RedirectRule struct {
+	Source string `json:"source"`
+	Target string `json:"target"`
+	Status string `json:"status"`
+}
+
+func getAvailableVersions() ([]string, error) {
+	// scans the website content directory to find available for wildcard redirect expansion
+	contentDir := "website/content/en"
+	entries, err := os.ReadDir(contentDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read content directory: %v", err)
+	}
+	var versions []string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			name := entry.Name()
+			versions = append(versions, name)
+		}
+	}
+	return versions, nil
+}
+
+func getRule(source string, target string, versions []string) []RedirectRule {
+	var rules []RedirectRule
+
+	if !strings.Contains(source, "*") {
+		return []RedirectRule{{
+			Source: source,
+			Target: target,
+			Status: "301",
+		}}
+	}
+
+	// Expand wildcard for each version
+	for _, version := range versions {
+		if strings.Contains(target, version) { // ensure non-looping redirects
+			continue
+		}
+		expandedSource := strings.ReplaceAll(source, "*", version)
+		rules = append(rules, RedirectRule{
+			Source: expandedSource,
+			Target: target,
+			Status: "301",
+		})
+	}
+
+	return rules
+}
+
+func main() {
+	redirectsFile := "website/static/_redirects"
+
+	if _, err := os.Stat(redirectsFile); os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "Error: %s not found\n", redirectsFile)
+		os.Exit(1)
+	}
+
+	versions, err := getAvailableVersions()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: Could not get available versions: %v\n", err)
+		os.Exit(1)
+	}
+
+	file, err := os.Open(redirectsFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading %s: %v\n", redirectsFile, err)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	var rules []RedirectRule
+	scanner := bufio.NewScanner(file)
+	lineNum := 0
+
+	for scanner.Scan() {
+		lineNum++
+		line := strings.TrimSpace(scanner.Text())
+
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		parts := strings.Fields(line)
+		if len(parts) == 2 {
+			expandedRules := getRule(parts[0], parts[1], versions)
+			rules = append(rules, expandedRules...)
+		} else {
+			fmt.Fprintf(os.Stderr, "Error: Invalid redirect format on line %d: %s\n", lineNum, line)
+			os.Exit(1)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading %s: %v\n", redirectsFile, err)
+		os.Exit(1)
+	}
+	jsonData, err := json.Marshal(rules)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
+		os.Exit(1)
+	}
```

Comment:
Why do we need to this part? Why can't we just print the RedirectRules as a json string?